src/foam/nanos/export/GoogleApiAuthService.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+ 
+package foam.nanos.export;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.extensions.java6.auth.oauth2.AuthorizationCodeInstalledApp;
+import com.google.api.client.extensions.jetty.auth.oauth2.LocalServerReceiver;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.FileDataStoreFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+
+public class GoogleApiAuthService {
+  private static final String CREDENTIALS_FOLDER = "NANOPAY_HOME";
+  private static final String CREDENTIALS_FILE = "credentials.json";
```

Comment:
How do the credentials work?

Reply:
We download file with credentials from google console, and then use them for retrieving access token for api, which we registered with google console.

Reply:
'download' - It looks like it downloads into NANOPAY_HOME - does this work when running the -u option? running from jar file?

Reply:
No, I did it manually. I wasn't sure what's the best way to automate it. Any suggestions?

Reply:
@jlhughes Never mind, I need a couple hours to figure it out

Reply:
so /tmp is available everywhere. but I it is possible to use the browser db or file system?

Reply:
NANOPAY_HOME changed to /tmp.
Hm, but browser db is used for storing client data if I'm not mistaken, while credentials.json is a server side data. Please let me know if I misunderstood.
I've read https://cloud.google.com/docs/authentication/production, and I still believe we need to store credentials somewhere locally

Reply:
ok, you can read/write to the 'var' directory.  If you look at the BMO planners/processors you'll see them managing temporary files in opt/nanopay/var.    
But in general we need a way to configure credentials on a running app - and then you can write the credentials.json locally if needed so it can be picked up the google library.